stone/backends/python_rsrc/stone_base.py

```diff
@@ -34,8 +34,53 @@ def __deepcopy__(self, memo):
         # disable copying so we can do identity comparison even after copying stone objects
         return self
 
+
 NOT_SET = NotSet()  # dummy object to denote that a field has not been set
 
+NO_DEFAULT = object()
+
+
+class Attribute(object):
+    __slots__ = ("name", "default", "nullable", "user_defined")
+
+    def __init__(self, name, nullable=False, user_defined=False):
```

Comment:
Brainstorming: What about a separate descriptor class for nullable attributes? This would make the `__get__` and `__set__` methods slightly faster.

Another idea would be to provide the validator as an attribute of `Attribute`, instead of putting it in the class dictionary. This way we could benefit from `__slots__`, and we wouldn't need to use `getattr` to access it.

Reply:
> What about a separate descriptor class for nullable attributes? This would make the `__get__` and `__set__` methods slightly faster.

I was also thinking about this, but it probably not worth it. Also again, we can always add it later if we will see any performance issues.
> Another idea would be to provide the validator as an attribute of `Attribute`

This is a good idea, it will probably not save huge memory but looks cleaner anyway.